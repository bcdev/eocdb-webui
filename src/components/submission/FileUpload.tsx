import * as React from 'react';
import { DropzoneArea } from 'material-ui-dropzone';
import { Theme, WithStyles } from "@material-ui/core";
import createStyles from "@material-ui/core/styles/createStyles";
import withStyles from "@material-ui/core/styles/withStyles";


// noinspection JSUnusedLocalSymbols
const styles = (theme: Theme) => createStyles({
    root: {}
});


interface FileUploadProps extends WithStyles<typeof styles> {
    label: string;
    onChange: (acceptedFiles: File[]) => void;
    files: File[];
    acceptedFiles?: string[];
    onDropRejected: (files: File[]) => void;
}


class FileUpload extends React.Component<FileUploadProps> {
    constructor(props: FileUploadProps) {
        super(props);
    }

    handleOnchange = (files: File[]) => {
        this.props.onChange(files);
    };

    handleOnDropRejected = (files: File[]) => {
       this.props.onDropRejected(files);
    };

    render() {
        return (
            <DropzoneArea
                onDropRejected={this.handleOnDropRejected}
                onChange={this.handleOnchange}
                filesLimit={500}
                maxFileSize={1000000000}
                showFileNamesInPreview={true}
                acceptedFiles={this.props.acceptedFiles}
                clearOnUnmount={true}
                dropzoneText={this.props.label}
            />
        );
    }
}

export default withStyles(styles)(FileUpload)