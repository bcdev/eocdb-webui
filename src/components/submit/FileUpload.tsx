import * as React from 'react';
import Dropzone from 'react-dropzone'
import Typography from "@material-ui/core/Typography/Typography";
import FileCopy from '@material-ui/icons/FileCopy';


interface FileUploadProps {
    onDrop: (acceptedFiles: any, rejectedFiles: any) => void;
    files: File[];

}

export class FileUpload extends React.Component<FileUploadProps> {
    constructor(props: FileUploadProps){
        super(props);
    }

    render() {
        const num_files = this.props.files.length;
        let fns = [];

        for (let i = 0; i < num_files; i++) {
            fns.push(
                <Typography key={this.props.files[i].name}>
                    <FileCopy/>
                    {this.props.files[i].name}
                </Typography>
            )
        }

        return (
            <Dropzone
                onDrop={this.props.onDrop}
            >
                {fns}
            </Dropzone>
        );
    }
}