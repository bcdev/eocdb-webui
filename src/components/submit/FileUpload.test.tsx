import * as React from 'react';
import * as renderer from 'react-test-renderer';
import FileUpload from "./FileUpload";


interface MockProps {
    onChange: () => void;
    files: File[];
}


const props: MockProps = {
    onChange: jest.fn(),
    files: [],
};


describe('FileUpload', () => {
    it('FileUpload renders', () => {
        const tree = renderer.create(
            <FileUpload {...props} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});