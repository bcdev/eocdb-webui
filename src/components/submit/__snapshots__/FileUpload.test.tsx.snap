// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload FileUpload renders 1`] = `
<div
  aria-disabled={false}
  className="DropzoneArea-dropZone-2"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="DropzoneArea-dropzoneTextStyle-5"
  >
    <p
      className="DropzoneArea-dropzoneParagraph-7"
    >
      Drag and drop an image file here or click
    </p>
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-8 DropzoneArea-uploadIconSize-6"
      color={undefined}
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
      />
    </svg>
  </div>
  <input
    accept="image/*,video/*,application/*"
    autoComplete="off"
    disabled={false}
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;
